src/AEG/CutNode.ts

```diff
@@ -315,4 +351,57 @@ export class CutNode {
         }
         return formulaString;
     }
+
+    /**
+     * Method that checks if a cut node is equal to another cut node. The are equal if they have
+     * the same children, irrespective of the ordering of nodes within a level
+     * @param otherCut The other cut node we are checking against for equality
+     * @returns True, if they are equal (the same). Else, false
+     */
+    public isEqualTo(otherCut: CutNode): boolean {
```

Comment:
We could get this down to an O(n log n) algo by sorting and comparing atoms on each layer, but your O(n^2) should be fine here since trees are going to be pretty small.